dart/lib/src/protocol/sentry_geo.dart

```diff
@@ -0,0 +1,29 @@
+/// Geographical location of the end user or device.
+class SentryGeo {
```

Comment:
Documented [here](https://develop.sentry.dev/sdk/event-payloads/types/#user) but not [here](https://develop.sentry.dev/sdk/event-payloads/user/).
I will check if this is inferred by the `ip_address` on the server side or if SDKs can actually set it as well.

Reply:
This is inferred from `ip_address` in the server, but if its [set](https://github.com/getsentry/relay/blob/637894b95b99e04403dd66c6d29a718a559c76ca/relay-general/src/store/normalize.rs#L826), it keeps as it is, I'm checking if SDKs should be allowed to set their own `geo` data.

Reply:
[Relay schema](https://github.com/getsentry/relay/blob/95f154409f236ecc8089a4b19534997625a21735/relay-general/src/protocol/user.rs#L9-L19)

Reply:
Let's keep it, I will update the develop docs.
We have to fix the Native bridges, to also sync the `geo` data.
Such as Android: https://github.com/getsentry/sentry-dart/blob/e390cedf364c65e61623aaffbdb9359bd3e090bf/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt#L265-L291
iOS https://github.com/getsentry/sentry-dart/blob/e390cedf364c65e61623aaffbdb9359bd3e090bf/flutter/ios/Classes/SentryFlutterPluginApple.swift#L503-L528

Reply:
Will do, but why is it this cumbersome to set the user on the native layer?

Reply:
Why cumbersome?

Reply:
I would have expected to be able to create a `User` out of the user json map

Reply:
Yeah, the native SDKs don't need it since deserialization never happens this way on the native side.
It'd be a nice addition tho, because this code is likely being repeated on every Hybrid SDK.
I'll create an issue for that, let's fix this on Flutter for now.

Reply:
https://github.com/getsentry/team-mobile/issues/59

Reply:
Both, Cocoa and Java, don't have the `Geo` object yet. I've added the data sync though

Reply:
https://github.com/getsentry/team-mobile/issues/59#issuecomment-1272832925
Added a comment on the `geo` field on Dart that this is not going to be synced for now (scope sync).

Reply:
Or maybe we add it under `data` for now?

Reply:
https://github.com/getsentry/develop/pull/705